### PR TITLE
MMT-2148 Improve UX on variable association page

### PR DESCRIPTION
**Pull request by ryanmiller-1**

### CODE REVIEW

File: app/controllers/collection_associations_controller.rb [hunk 1]
```diff
@@ -26,21 +26,36 @@ def index
...
       association_count = association_response.body['hits']
       association_results = association_response.body['items']
+    else
+      association_count = 0
+      association_results = []
+      Rails.logger.error("An error occurred while trying to fetch collection associations for a #{lower_resource_name}.  CMR Response: #{association_response.clean_inspect}")
+      flash[:error] = "An error occurred while trying to retrieve associations for this #{lower_resource_name}. Please try again before contacting #{view_context.mail_to('support@earthdata.nasa.gov', 'Earthdata Support')}"
     end
 
+    @can_associate = not_variable? || association_results.blank?
```
Comment (ryanmiller-1) [inline app/controllers/collection_associations_controller.rb:36]: add comment here

File: app/controllers/collection_associations_controller.rb [hunk 2]
```diff
@@ -26,21 +26,36 @@ def index
...
     @associations = Kaminari.paginate_array(association_results, total_count: association_count).page(page).per(RESULTS_PER_PAGE)
   end
 
   def new
+    @previously_associated_collections = get_all_collection_associations
+    if variable? && @previously_associated_collections.present?
+      flash[:error] = "This #{lower_resource_name} already has a Collection Association. To change the association, you must first remove the existing collection association."
+      redirect_to send("#{lower_resource_name}_collection_associations_path", resource_id)
+      return
```
Comment (ryanmiller-1) [inline app/controllers/collection_associations_controller.rb:45]: removable?

File: app/controllers/collection_associations_controller.rb [hunk 3]
```diff
@@ -26,21 +26,36 @@ def index
...
+      flash[:error] = "An error occurred while trying to retrieve associations for this #{lower_resource_name}. Please try again before contacting #{view_context.mail_to('support@earthdata.nasa.gov', 'Earthdata Support')}"
     end
 
+    @can_associate = not_variable? || association_results.blank?
     @associations = Kaminari.paginate_array(association_results, total_count: association_count).page(page).per(RESULTS_PER_PAGE)
   end
 
   def new
+    @previously_associated_collections = get_all_collection_associations
+    if variable? && @previously_associated_collections.present?
```
Comment (ryanmiller-1) [inline app/controllers/collection_associations_controller.rb:42]: comment

File: app/controllers/collection_associations_controller.rb [hunk 4]
```diff
@@ -26,21 +26,36 @@ def index
...
     add_breadcrumb 'New', send("new_#{lower_resource_name}_collection_association_path", resource_id)
 
     super
-
-    @previously_associated_collections = get_all_collection_associations
   end
 
   def create
-    association_response = cmr_client.send("add_collection_assocations_to_#{lower_resource_name}", resource_id, params[:selected_collections], token)
+    association_response = if variable?
```
Comment (ryanmiller-1) [inline app/controllers/collection_associations_controller.rb:54]: comment

File: app/views/collection_associations/index.html.erb [hunk 5]
```diff
@@ -2,9 +2,11 @@
   <section>
     <h2><%= @resource['Name'] %> Collection Associations</h2>
 
-    <%= link_to 'Add Collection Associations', send("new_#{lower_resource_name}_collection_association_path", resource_id), class: 'eui-btn--blue space-bot' %>
+    <% if @can_associate %>
+      <%= link_to 'Add Collection Associations', send("new_#{lower_resource_name}_collection_association_path", resource_id), class: 'eui-btn--blue space-bot' %>
+    <% end %>
 
-    <% if @associations.any? %>
+    <% if @associations.any? && not_variable? %>
```
Comment (ryanmiller-1) [inline app/views/collection_associations/index.html.erb:9]: remove

File: app/controllers/collection_associations_controller.rb [hunk 6]
```diff
@@ -26,21 +26,42 @@ def index
...
-
-    @previously_associated_collections = get_all_collection_associations
   end
 
   def create
-    association_response = cmr_client.send("add_collection_assocations_to_#{lower_resource_name}", resource_id, params[:selected_collections], token)
+    # The form is slightly different for variables. Despite only accepting one
+    # association, the collection id for variables still needs to be in an array
+    association_response = if variable?
+                             cmr_client.send("add_collection_assocations_to_#{lower_resource_name}", resource_id, [params[:selected_collection]], token)
```
Comment (charleshuang80) [inline app/controllers/collection_associations_controller.rb:61]: I think `Array.wrap` would be preferred here

File: app/views/collection_associations/new.html.erb [hunk 7]
```diff
@@ -3,8 +3,14 @@
...
+    <% if not_variable? %>
+      <p><span class="fa fa-info-circle"></span> Disabled rows in the results below represent collections that are already associated with this <%= lower_resource_name %>.</p>
 
-    <%= render partial: 'cmr_search/shared/search_results', locals: { collections: @collections, ignored_concept_ids: @previously_associated_collections, submit_path: send("#{lower_resource_name}_collection_associations_path", @concept_id), submit_text: 'Submit' } %>
+      <%= render partial: 'cmr_search/shared/search_results', locals: { collections: @collections, ignored_concept_ids: @previously_associated_collections, submit_path: send("#{lower_resource_name}_collection_associations_path", @concept_id), submit_text: 'Submit' } %>
+    <% else %> 
+      <p><span class="fa fa-info-circle"></span> Variables can only be associated with a single collection.</p>
+
+      <%= render partial: 'cmr_search/shared/search_results', locals: { collections: @collections, ignored_concept_ids: @previously_associated_collections, submit_path: send("#{lower_resource_name}_collection_associations_path", @concept_id), submit_text: 'Submit', use_radio_buttons: true } %>
```
Comment (charleshuang80) [inline app/views/collection_associations/new.html.erb:13]: I think you can just call the render partial once, and use `variable?` or `not_variable?` for the value for `use_radio_buttons`

**Pull request merged**